tests/phpunit/tests/test-links-domain-sub-dir.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+class Links_Domain_Sub_Dir_Test extends PLL_Domain_UnitTestCase {
+	public function set_up() {
+		parent::set_up();
+
+		$this->hosts = array(
+			'en' => 'http://example.org',
+			'fr' => 'http://example.fr',
+			'de' => 'http://example.de',
+		);
+
+		self::$model->options['hide_default'] = 1;
+		self::$model->options['force_lang'] = 3;
+		self::$model->options['domains'] = $this->hosts;
+
+		$this->init_links_model();
+	}
+
+	public function test_flags_urls_sub_dir() {
+		// Fake WP install in subdir.
+		update_option( 'siteurl', 'http://example.org/sub' );
+		update_option( 'home', 'http://example.org' );
+
+		// Filter `plugins_url` because `WP_CONTENT_URL` is already defined...
+		add_filter(
+			'plugins_url',
+			function( $url ) {
+				$url = str_replace( POLYLANG_DIR . '/', '/polylang/', $url );
+
+				return str_replace( 'wp-content', 'sub/wp-content', $url );
+			},
+			-1
+		);
```

Comment:
Can be done in one call:
```suggestion
		add_filter(
			'plugins_url',
			function( $url ) {
				return str_replace(
					array( POLYLANG_DIR . '/', 'example.org' ),
					array( '/polylang/', 'example.org/sub' ),
					$url
				);
			},
			-1
		);
```

Note:
It could also go in `PLL_Domain_UnitTestCase::set_up()` in this form:
```php
$this->orig_siteurl = get_option( 'siteurl' );

add_filter(
	'plugins_url',
	function ( $url ) {
		$siteurl = get_option( 'siteurl' );

		if ( false === strpos( $url, $siteurl ) ) {
			$url = str_replace( $this->orig_siteurl, $siteurl, $url );
		}
		return str_replace( POLYLANG_DIR . '/', '/polylang/', $url );
	},
	-1
);
```

Reply:
Good idea! Moved into a trait in [713b0ca](https://github.com/polylang/polylang/pull/1300/commits/713b0cac724bf743c87013541742eb15b03a6863).